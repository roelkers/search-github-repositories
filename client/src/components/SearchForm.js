import React from 'react';

class SearchForm extends React.Component {
  state = {
    searchTerm: '',
  };

  handleInputChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  handleSubmit = e => {
    e.preventDefault();
    if (this.state.searchTerm.trim()) {
      this.props.onSearchRepository(this.state.searchTerm);
      this.searchTerm = '';
    }
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit} className='searchForm'>
        <input
          type="text"
          placeholder="repository name"
          name="searchTerm"
          onChange={this.handleInputChange}
          value={this.state.searchTerm}
        />
        <button type="submit">Search</button>
      </form>
    );
  }
}

export default SearchForm;