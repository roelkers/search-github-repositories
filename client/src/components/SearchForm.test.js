import React from 'react';
import SearchForm from './SearchForm';
import renderer from 'react-test-renderer';

describe('SearchForm', () => {

    it('should match the snapshot', () => {
        const component = renderer.create(<SearchForm />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});

//TO DO : form submit test
