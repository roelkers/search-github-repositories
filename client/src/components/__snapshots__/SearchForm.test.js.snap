// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<form
  className="searchForm"
  onSubmit={[Function]}
>
  <input
    name="searchTerm"
    onChange={[Function]}
    placeholder="repository name"
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
